ui-modules/blueprint-composer/app/components/catalog-saver/catalog-saver.directive.js

```diff
@@ -68,6 +68,30 @@ export function saveToCatalogModalDirective($rootScope, $uibModal, $injector, $f
         }
     }
 
```

Comment:
I would not include so many custom calls from the override.
 
Suggestion:

```
export function saveToCatalogModalDirective($rootScope, $uibModal, $injector, $filter, composerOverrides, blueprintService) {
    let modalDirective = this;

    modalDirective.getConfigNameFromEntity = (entity) => {
    
    // use $scope directly
    // code omitted ...

    };

    modalDirective.getConfigBundleNameFromScope = () => {

    // use $scope directly
    // code omitted ...

    };

    modalDirective.getSuggestedSymbolicNameFromBlueprint = (entity, metadata) => {

    // use $scope directly
    // code omitted ...

    };

    modalDirective.updateCatalogConfig = (config, $element) => {

    // use $scope directly
    // code omitted ...

    }

    // allow downstream to configure directive and scope here
    (composerOverrides.configureCatalog || () => {})(modalDirective, $scope);

    return = {

    // code omitted ...
```

Customize methods defined on `modalDirective` instance in the branding part and call them accessing instance of the directive `modalDirective.getConfigNameFromEntity(entity)`

Reply:
Correction: it is better to define methods above at the controller level, not at directive:
```
export function saveToCatalogModalDirective($rootScope, $uibModal, $injector, $filter, composerOverrides, blueprintService) {
    return {
        restrict: 'E',
        templateUrl: function (tElement, tAttrs) {
            return tAttrs.templateUrl || TEMPLATE_URL;
        },
        scope: {
            config: '=',
        },
        controller: ['$scope','composerOverrides', saveToCatalogModalController],
        link: link
    };

    function saveToCatalogModalController($scope, composerOverrides) {
        let modalController = this;

        modalController.getConfigNameFromEntity = (entity) => {

            // use $scope directly
            // code omitted ...

        };
        
        // etc. ...

        // allow downstream to configure directive and scope here
        (composerOverrides.configureCatalogModal || function () {})(modalController, $scope);
    }
```

Reply:
Tried it, the pattern cannot be applied to this implementation.